### PR TITLE
Tbrands 97 featured image

### PR DESCRIPTION
**Pull request by JackHowa**
## Description

Add product featured image 

## Jira Ticket

- [TBRANDS-97](https://arcpublishing.atlassian.net/browse/TBRANDS-97)

## Acceptance Criteria

block migration ac
- [x] add test coverage for image mock data
- [x] Write readme to include purpose
- [x] Update storybook with examples 
- [x] Take in auth token. Set the variable that is getting this environment to be RESIZER_APP_VERSION to be 2 for this feature pack and this mock data
- [x] Set the alt text in the mock image and pull that down. If no alt, then use the fallback in the image component

acceptance criteria ticket 
- [x] Product Featured Image - Arc Block is available to be added to a page in Page Builder Editor.
- [x] The Product Featured Image is available for display from the Arc Commerce Content Source (TBRANDS-94)
- [x] This block will only output the Product Featured Image only.
- [x] Product Featured Image - Arc Block display and styling match designs.
- [x] Product Featured Image - Arc Block is responsive and works across all device types.

## Test Steps

1. Checkout this branch `git checkout TBRANDS-97-featured-image`
2. See feature pack pr https://github.com/WPMedia/arc-themes-feature-pack/pull/311
3. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/product-featured-image-block`
4. Create a layout of the right rail on commerce 
5. Add the featured image block 

### Before

- block not available 

### After

with right rail commerce 

desktop: 
Screen Shot 2022-07-28 at 13 20 40

mobile 
Screen Shot 2022-07-28 at 13 20 59

## Dependencies or Side Effects

- Goes with https://github.com/WPMedia/arc-themes-feature-pack/pull/311
-  manifest https://github.com/WPMedia/arc-themes-manifests/pull/118

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/product-featured-image-block/index.story.jsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// for more info on storybook and using the component explorer
+// https://storybook.js.org/
+export default {
+	title: "Blocks/Product Featured Image",
+	parameters: {
+		chromatic: { viewports: [320, 1200] },
+	},
+};
+
+export const showHeading = () => <ProductFeaturedImage customFields={{ showHeading: true }} />;
```
Comment (JackHowa) [inline blocks/product-featured-image-block/index.story.jsx]: sb is not updated

File: blocks/product-featured-image-block/features/product-featured-image/default.test.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		const { container } = render(<ProductFeaturedImage />);
+		expect(container.firstChild).toBe(null);
+	});
+	it("should render a featured image", () => {
+		useFusionContext.mockImplementation(() => ({
+			globalContent: {
+				schema: {
+					featuredImage: {
+						value: {
+							assets: [FEATURED_IMAGE_ASSET],
```
Comment (JackHowa) [inline blocks/product-featured-image-block/features/product-featured-image/default.test.jsx:62]: not a lot of fallback logic if there's no schema of featuredImage. Possibly add test cases if we're not getting the schema we're expecting

File: blocks/product-featured-image-block/features/product-featured-image/default.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+export const ProductFeaturedImageDisplay = ({ data, resizerAppVersion }) => {
+	const featuredImageAsset = data.schema?.featuredImage?.value?.assets.find(
+		(asset) => asset.type === "image"
+	);
+
+	if (!featuredImageAsset) {
+		return null;
+	}
+
+	const { url, auth, alt_text: altText } = featuredImageAsset;
```
Comment (matthewroach) [inline blocks/product-featured-image-block/features/product-featured-image/default.jsx:18]: `alt_text` may not always be present on the data, are we accounting for that?

Comment (JackHowa) [reply blocks/product-featured-image-block/features/product-featured-image/default.jsx:18]: the Image component is. But I can add a test to validate it if we want.

Comment (JackHowa) [reply blocks/product-featured-image-block/features/product-featured-image/default.jsx:18]: updated

**Review by nschubach — APPROVED** [review]
Seems to be working per AC/tests... (updated link to ticket as it's not TMEDIA-97) ;)

**Comment by JackHowa** [comment]
@haganka ready for design qa!

**Pull request merged**